src/main/java/io/jenkins/pluginhealth/scoring/service/PluginService.java

```diff
@@ -37,6 +39,24 @@ public PluginService(PluginRepository pluginRepository) {
         this.pluginRepository = pluginRepository;
     }
 
+    public void updateDatabase(Plugin plugin) {
+        Optional<Plugin> pluginFromDatabase = pluginRepository.findByName(plugin.getName());
```

Comment:
Using `Optional` is a good idea but you are not using it to its potential. Look at the `map` and `orElse` method. You could reduce the number of lines using those, without compromising the readability of the code.

Reply:
Sure, I'm on it now

Reply:
[Correct me if I'm wrong]: As far as I could understand and based on the currently pushed code, the end result should be something like:
`pluginFromDatabase.map()`
`            .orElse(pluginRepository.save(plugin));`

But I'm not able to understand how to use the `map()` method. Can you please help?

Reply:
we could do something like

```java
pluginRepository.findByName(plugin.getName())
  .map(store -> store.withSCM(plugin.getSCM()).withReleaseTimestamp(plugin.getTimestamp()))
  .map(updated -> pluginRepository.save(updated)) // this can be a changed by a reference call
  .orElse(pluginRepository.save(plugin));
```

Reply:
Is this `withSCM()` and `withReleaseTimestamp` related to this: https://projectlombok.org/features/With?

Reply:
No, please don't add Lombok on the project. We don't need this. 
It's just the setters, in the "builder" format, so they return the instance. This allow us to link the calls.

Reply:
[Correct me if I'm wrong]: I read [this article](https://www.geeksforgeeks.org/builder-pattern-in-java/) and I modified the setter methods to return a `Plugin` instance so that we can chain multiple setter methods to eventually return a `Plugin` object instance.